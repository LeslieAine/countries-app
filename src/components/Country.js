import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from '../styles/Country.module.css';

const Country = ({ name, flagEmoji, continent }) => (
  <Link to={`/country/${name}`} className={styles.link}>
    <li className={styles.country}>
      <h2 className={styles.countryName}>
        {name}
        <br />
        {flagEmoji}
      </h2>
      <p>
        {continent}
      </p>
    </li>
  </Link>
);

Country.propTypes = {
  name: PropTypes.string.isRequired,
  flagEmoji: PropTypes.string.isRequired,
  continent: PropTypes.string.isRequired,
};

export default Country;
