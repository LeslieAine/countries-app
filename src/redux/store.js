import countriesReducer from './countries/countriesSlice';

const { configureStore } = require('@reduxjs/toolkit');

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
