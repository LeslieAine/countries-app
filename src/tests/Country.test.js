import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Country from '../components/Country';

describe('Country', () => {
  it('should render the component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Country name="Republic of Guatemala" flagEmoji="GT" continent="North America" />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
