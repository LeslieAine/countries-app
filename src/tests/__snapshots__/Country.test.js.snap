// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country should render the component 1`] = `
<a
  className="link"
  href="/country/Republic of Guatemala"
  onClick={[Function]}
>
  <li
    className="country"
  >
    <h2
      className="countryName"
    >
      Republic of Guatemala
      <br />
      GT
    </h2>
    <p>
      North America
    </p>
  </li>
</a>
`;
